### PR TITLE
supermarket release checklist documentation

### LINKED ISSUE #2206: Release Supermarket 4.x
We have a HUGE amount of new functionality and fixes in supermarket main branch right now that we need to ship to customers:

https://github.com/chef/supermarket/wiki/Pending-Release-Notes

Let's get this out the door, but first we need to validate the builds function as expected:

- [x]  Installs work and Supermarket is functional post install

- [x] Upgrades from 3.4.8 -> 4.x work and Supermarket is functional post upgrade 

- [x] Users can login using Infra Server users and view their profile

- [x] Cookbooks can be uploaded via knife

- [x] Cookbooks can be downloaded via `chef install` command from a Policyfile

- [x] Quality metrics are run on all uploaded cookbooks

If there's any other potential validation steps let's get them added here or in a dev doc.

### PR DESCRIPTION
**Pull request by RajeshPaul38**
### Description

Added documentation for supermarket release checklist.

### Issues Resolved

#2206 

### Check List

- [ ] New functionality includes tests
- [ ] All buildkite tests pass
- [ ] Full omnibus build and tests in buildkite pass
- [X] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/main/CONTRIBUTING.md#developer-certification-of-origin-dco>
- [ ] PR title is a worthy inclusion in the CHANGELOG

### CODE REVIEW

File: docs/supermarket-release-checklist.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
+### Testing on ubuntu 20.04.2 LTS
+
+- Build supermarket artifact with omnibus
+  - Go to the supermarket repo. Inside repo go to `omnibus` directory
```
Comment (dheerajd-msys) [inline docs/supermarket-release-checklist.md:4]:
```suggestion
  - Go to the supermarket repo. Inside repo go to `omnibus` directory
`cd omnibus`
```

File: docs/supermarket-release-checklist.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
+### Testing on ubuntu 20.04.2 LTS
+
+- Build supermarket artifact with omnibus
+  - Go to the supermarket repo. Inside repo go to `omnibus` directory
+  - run `bundle install --binstubs`
+
+- Install supermarket artifact
+  - Once build is successful it will create a supermarket installation package in the directory `/var/cache/omnibus/pkg`.
+    U can install the package using respective package installer for the selected OS.
+
```
Comment (dheerajd-msys) [inline docs/supermarket-release-checklist.md:10]: Could be better if we provide command based instruction here e.g. `dpkg -I /var/cache/omnibus/pkg/<pkg>` as we are providing instruction for Ubuntu

File: docs/supermarket-release-checklist.md [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+Had to set the same IP in the redirect url in chef server oc-id applications as follows:
+```
+"redirect_uri": "https://<ec2 IP address>/auth/chef_oauth2/callback"
+```
+
+Now it's able to authorize supermarket with oc-id. But the only issue remains is that it's saying the certificate in supermarket is self signed and hence not setting the authorization token in cookie. Will check what can be done.
+
+Now as the self signed certificate issue is resolved we move on to the next step.
+
```
Comment (dheerajd-msys) [inline docs/supermarket-release-checklist.md:36]: We can mention how `self signed certificate issue` is resolved.

**Review by dheerajd-msys — APPROVED** [review]

**Comment by dheerajd-msys** [comment]
Looks good overall !! Inline comments added only for readability.

File: docs/supermarket-release-checklist.md (hunk 2 above)
Comment (RajeshPaul38) [reply docs/supermarket-release-checklist.md:13]: > Could be better if we provide command based instruction here e.g. `dpkg -I /var/cache/omnibus/pkg/<pkg>` as we are providing instruction for Ubuntu

Added the command.

**Comment by RajeshPaul38** [comment]
@dheerajd-msys have updated the doc as per the review feedbacks and also added few needed clarifications for better understanding of the reasoning behind any fix.

**Pull request merged**